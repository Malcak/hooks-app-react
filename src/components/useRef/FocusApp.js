import React, { useRef } from 'react';

export const FocusApp = () => {
  const inputRef = useRef();

  const handleClick = () => {
    inputRef.current.select();
  };

  return (
    <div>
      <h1>Focus Screen</h1>
      <hr />
      <input
        className="form-control mb-1"
        placeholder="Your name"
        ref={inputRef}
      />
      <button className="btn btn-outline-primary" onClick={handleClick}>
        Focus
      </button>
    </div>
  );
};
