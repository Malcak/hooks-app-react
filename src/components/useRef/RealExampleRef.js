import React, { useState } from 'react';
import { MultipleCumtomHooks } from '../multipleCustomHooks/MultipleCumtomHooks';

export const RealExampleRef = () => {
  const [show, setShow] = useState(true);

  const handleClick = () => {
    setShow(!show);
  };

  return (
    <div>
      <h1>Real Example of useRef</h1>
      <br />
      {show && <MultipleCumtomHooks />}
      <button className="btn btn-primary" onClick={handleClick}>
        Show/Hide quotes
      </button>
    </div>
  );
};
