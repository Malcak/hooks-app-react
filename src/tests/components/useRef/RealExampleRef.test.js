import React from 'react';
import { shallow } from 'enzyme';
import { RealExampleRef } from '../../../components/useRef/RealExampleRef';

describe('tests on the <RealExampleRef /> component', () => {
  let wrapper = shallow(<RealExampleRef />);
  beforeEach(() => {
    wrapper = shallow(<RealExampleRef />);
  });

  test('should display the component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('should change the status and display <MultipleCustomHooks />', () => {
    wrapper.find('button').simulate('click');
    expect(wrapper.find('MultipleCumtomHooks').exists()).toBe(true);
  });
});
