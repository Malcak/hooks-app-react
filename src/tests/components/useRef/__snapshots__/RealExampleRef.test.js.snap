// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on the <RealExampleRef /> component should display the component correctly 1`] = `
<div>
  <h1>
    Real Example of useRef
  </h1>
  <br />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Show/Hide quotes
  </button>
</div>
`;
